Makefile

```diff
@@ -14,24 +14,30 @@ done)
 
 GOARCH=$(shell go env GOARCH)
 HOST_ARCH=$(shell arch)
-SKIP_GO_VERSION_CHECK=
+ifeq ($(GOPATH),)
+    SKIP_GO_VERSION_CHECK=y
+else
+    SKIP_GO_VERSION_CHECK=
+endif
 
 ifeq ($(SKIP_GO_VERSION_CHECK),)
     include golang.mk
 endif
 
-ifeq ($(ARCH),)
-    ARCH = $(GOARCH)
-endif
+ifneq ($(GOPATH),)
+    ifeq ($(ARCH),)
+        ARCH = $(GOARCH)
+    endif
 
-ARCH_DIR = arch
-ARCH_FILE_SUFFIX = -options.mk
-ARCH_FILE = $(ARCH_DIR)/$(ARCH)$(ARCH_FILE_SUFFIX)
-ARCH_FILES = $(wildcard arch/*$(ARCH_FILE_SUFFIX))
-ALL_ARCHES = $(patsubst $(ARCH_DIR)/%$(ARCH_FILE_SUFFIX),%,$(ARCH_FILES))
+    ARCH_DIR = arch
```

Comment:
none of the following variables are dependent of GOPATH

Reply:
If they are moved out from "ifneq ($(GOPATH),)" will get following error without GOPATH:
sudo make install
[sudo] password for teawater:
Makefile:39: arch/-options.mk: No such file or directory
make: *** No rule to make target 'arch/-options.mk'.  Stop.